XCUITests/DataManagementTests.swift

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+class DataManagementTests: BaseTestCase {
+
+    //Testing the entries are correctly added and deleted. This a termporary behaviour. This test will be changed after releasing Firefox 15.x and Bug 1499074 will be fixed.
+    func testWebSiteDataEnterFirstTime() {
+        navigator.goto(WebsiteDataSettings)
+        XCTAssertEqual(app.tables.cells.count, 2)
+        XCTAssertTrue(app.tables.staticTexts["Clear All Website Data"].exists)
+        navigator.performAction(Action.AcceptClearAllWebsiteData)
+        XCTAssertEqual(app.tables.cells.count, 1)
+        navigator.createNewTab()
+        navigator.goto(WebsiteDataSettings)
+        XCTAssertEqual(app.tables.cells.count, 2)
+    }
+    
+    // Testing the search bar, search results and 'Show More' option.
+    func testWebSiteDataOptions() {
+        navigator.openURL("google.com")
+        navigator.openURL("youtube.com")
+        navigator.openURL("twitter.com")
+        navigator.openURL("facebook.com")
+        navigator.goto(WebsiteDataSettings)
+        navigator.performAction(Action.TapOnFilterWebsites)
+        app.typeText("google")
+        waitforExistence(app.tables["Search results"])
+        XCTAssertEqual(app.tables["Search results"].cells.count, 2)
+        navigator.performAction(Action.TapOnFilterWebsites)
+        app.searchFields["Filter Sites"].tap()
+        app.typeText("foo")
+        XCTAssertEqual(app.tables["Search results"].cells.count, 0)
+        app.buttons["Cancel"].tap()
+        app.tables/*@START_MENU_TOKEN@*/.staticTexts["Show More"]/*[[".cells[\"ShowMoreWebsiteData\"].staticTexts[\"Show More\"]",".staticTexts[\"Show More\"]"],[[[-1,1],[-1,0]]],[0]]@END_MENU_TOKEN@*/.tap()
+        XCTAssertNotEqual(app.tables.cells.count, 9)
+        navigator.performAction(Action.AcceptClearAllWebsiteData)
+        waitforExistence(app.tables.staticTexts["Clear All Website Data"])
```

Comment:
Here you could also use an ID: `app.tables.cells["ClearAllWebsiteData"]`
As a rule of thumb, use IDs whenever is possible. If there is not ID, and is difficult to add it, then use label as you did.